Common/src/geometry_structure.cpp

```diff
@@ -2792,6 +2794,90 @@ void CGeometry::UpdateCustomBoundaryConditions(CGeometry **geometry_container, C
   }
 }
 
+
+void CGeometry::ComputeSurf_Straightness(CConfig *config, bool print_on_screen) {
+
+  bool RefUnitNormal_defined;
+  unsigned short iMarker,
+                 iDim;
+  unsigned long iVertex,
+                iPoint;
+  su2double epsilon = 1.0e-6,
+            Area;
+
+  su2double *Normal        = new su2double[nDim],
+            *UnitNormal    = new su2double[nDim],
+            *RefUnitNormal = new su2double[nDim];
+
+  /*--- Allocate memory and assume the less restrictive, i.e. all boundaries are not straight ---*/
+  bound_is_straight = new bool [nMarker];
+  for (iMarker = 0; iMarker < nMarker; iMarker++) 
+    bound_is_straight[iMarker] = false;
+
+  /*--- Loop over all the markers ---*/
+  for (iMarker = 0; iMarker < nMarker; iMarker++) {
+    
+    if (config->GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE ||
+        config->GetMarker_All_KindBC(iMarker) == EULER_WALL) {
+      
+      /*--- Assume now that this boundary marker is straight. As soon as one
+        AreaElement is found that is not aligend with a Reference then it is 
+        certain that the boundary marker is not straight and one can stop 
+        searching ---*/
+      bound_is_straight[iMarker] = true;
+      RefUnitNormal_defined = false; 
+      iVertex = 0;
+
+      while(bound_is_straight[iMarker] == true && iVertex < nVertex[iMarker]) {
+        
+        iPoint = vertex[iMarker][iVertex]->GetNode();
+        
+        if (node[iPoint]->GetDomain()) {
+          
+          vertex[iMarker][iVertex]->GetNormal(Normal);
+          /*--- Compute unit normal. ---*/
+          Area = 0.0;
+          for (iDim = 0; iDim < nDim; iDim++)
+            Area += Normal[iDim]*Normal[iDim];
+          Area = sqrt (Area);
+          
+          for (iDim = 0; iDim < nDim; iDim++)
+            UnitNormal[iDim] = -Normal[iDim]/Area;
+          
+          /*--- Check if unit normal is within tolerance of the Reference unit normal. 
+                Reference unit normal = first unit normal found. ---*/
+          if(RefUnitNormal_defined) {
+            for (iDim = 0; iDim < nDim; iDim++) {
+              if( abs(RefUnitNormal[iDim] - UnitNormal[iDim]) > epsilon )
+                bound_is_straight[iMarker] = false; break;
+            }
+          } else {
+            for (iDim = 0; iDim < nDim; iDim++) RefUnitNormal[iDim] = UnitNormal[iDim];   
+            RefUnitNormal_defined = true;
+          }
+
+        }//if domain
+      iVertex++;
+      } //while iVertex
+
+    /*--- Print results on screen. ---*/
+    if(rank == MASTER_NODE && print_on_screen) {
```

Comment:
Yeah this one was a bit more involved because of the good'ol local vs global marker stuff. 

In this previous version each process had its own bound_is_straight array and accessed that in the BC_sym routine -> therefore the computation was correct i think. It would have happened that on the Master-process the value is `true` and on another `false` for the same global marker (e.g. picture the 2D wedge cases split in the middle), but as only the master-process prints its value to screen ...the info on screen would be wrong.

I changed it now: there is an `Allreduce` in the end such that each process has the same globally correct info. I acknowledge that it is prob not the most elegant implementation, so if there is a hint on how to do it better...

Reply:
MPI wise looks ok, idk if there is a better way in SU2 to get a iMarker to iMarkerGlobal correspondence than comparing strings though.
Maybe let each rank decide if its patch should be straight or not and communicate only the messages you want to print to master and don't do the reduction.
At least convert back from global to local in the preprocessing to avoid comparing strings inside the actual boundary condition.

Reply:
Reverted implementation back to local `vector<bool> bound_is_straight` .  But I kept a global version in the `geometry->ComputeSurf_Straightness` to print the result on screen.